test/cppsuite/src/main/test.cpp

```diff
@@ -135,12 +142,16 @@ test::run()
       db_create_config, _args.home.empty() ? DEFAULT_DIR : _args.home);
 
     /* Initiate the load stage of each component. */
-    for (const auto &it : _components)
-        it->load();
+    for (const auto &it : _components) {
+        if (it->enabled())
+            it->load();
+    }
 
     /* Spawn threads for all component::run() functions. */
-    for (const auto &it : _components)
-        _thread_manager->add_thread(&component::run, it);
+    for (const auto &it : _components) {
+        if (it->enabled())
+            _thread_manager->add_thread(&component::run, it);
+    }
 
```

Comment:
It appears that components that are not enabled, are not expected to be enabled during the test. Have you considered something like:

```suggestion
    // Assumes: using std;
    // Note: my c++ is rusty: assume there are syntax and other errors here
    // this is meant as a sketch.

    // Identify all enabled components.
    vector<_components::value_type> enabled{_components.size()};
    copy_if(_components.begin(), _components.end(), back_inserter(enabled),
        [](auto &c){ return c->enabled(); });
    
    // Each component runs in its own thread.
    for_each(_enabled.begin(), _enabled.end(), 
        [&_thread_manager](auto &c){ _thread_manager->add_thread(&component::run, c)});
    
    // This pattern can be adapted to subsequent loops.
```

Irrespective of this suggestion. L169 below calls `end_run()` on all components even those that are not enabled. If that is intentional, I think a comment explain as to why would be helpful.

Reply:
Thanks for the suggestion! As I was trying this implementation, I realised that I could do this somewhere else. Please, can you review 90dcb9c46d8a947b04cbd0d8835a216284d09e4c? I am now saving the components in array [here](https://github.com/wiredtiger/wiredtiger/pull/9711/commits/90dcb9c46d8a947b04cbd0d8835a216284d09e4c#diff-49394636c7b1a3f9f0097b0af2f37e7f993f544986f5c590b720032a843bbe7aR55-R59).